setup.py

```diff
@@ -111,6 +111,14 @@
             "pre-commit",
         ],
         "modin-development": [
+            f"snowflake-connector-python[pandas]{CONNECTOR_DEPENDENCY_VERSION}",
+            "pytest<8.0.0",  # check SNOW-1022240 for more details on the pin here
+            "pytest-cov",
+            "coverage",
+            "sphinx==5.0.2",
+            "cachetools",  # used in UDF doctest
+            "pytest-timeout",
+            "pre-commit",
```

Comment:
let's put the `development` requirements in a variable and reuse them instead of copying them